src/main/webapp/WEB-INF/view/profile.jsp

```diff
@@ -24,36 +29,55 @@ UserStore userStore = UserStore.getInstance();
       String currentUser = (String) request.getSession().getAttribute("user");
       String profileUser = (String) request.getAttribute("profilePage");
       User thisUser = userStore.getUser(profileUser);
+
       if (profileUser.equals("")) { %>
         <h1>User does not exist.</h1>
 
-      <% } else if (currentUser != null && currentUser.equals(profileUser)) { %>
-        <h2 style="color:blue">Welcome to your page!</h2>
+      <% } else {
+        if (currentUser != null && currentUser.equals(profileUser)) { %>
+            <h1 style="color:dodgerblue">Welcome to your page!</h1>
+        <% } else { %>
+            <h1 style="color:dodgerblue">Welcome to <%= profileUser %>'s Page!</h1>
+        <% } %>
         <% /** Gets the bio of this user to display on their profile page */ %>
-        <% String profilePageBio = thisUser.getBio(); %>
-        <h3 style="color:purple">About Me</h3>
-        <a> <%= profilePageBio %> </a>
-        <br/>
-        <br/>
-
-        <% /** Gives current user a form that allows them to edit their own bio */ %>
-        <a> Edit your bio here! (only you can see this) </a>
-        <form action="/users/<%= currentUser %>" method="POST">
-            <input type="text" name="bio" value="<%= thisUser.getBio() %>" >
+            <% String profilePageBio = thisUser.getBio(); %>
+            <h2 style="color:skyblue">About Me</h2>
+            <a> <%= profilePageBio %> </a>
             <br/>
-            <button type="submit">Submit</button>
-        </form>
+            <br/>
+        <% if (currentUser != null && currentUser.equals(profileUser)) { %>
+            <% /** Gives current user a form that allows them to edit their own bio */ %>
+                <a> Edit your bio here! (only you can see this) </a>
+                <form action="/users/<%= currentUser %>" method="POST">
+                    <input type="text" name="bio" value="<%= thisUser.getBio() %>" >
+                    <br/>
+                <button type="submit">Submit</button>
+                </form>
+         <% } %>
+         <h2 style="color:skyblue"> <%= profileUser %>'s Sent Messages </h2>
+         <% List<Message> userMessages = (List) request.getAttribute("messages");
+            Instant instant = null;
+            LocalDateTime localDate = null;
+            for (Message m: userMessages) {
```

Comment:
It would probably be nice to have this whole process done in its own separate method, that way would could use the same format for the conversations and activity feed, which would really come in handy

Reply:
That's a good idea, I can make a class that converts an Instant object into a simple String of the time, and just put the code there.

Reply:
Yes! That would be very helpful, and would give us a centralized location for that process in case we want to make sitewide changes to how it works.